cmd/benchmark.go

```diff
@@ -1,267 +1,81 @@
-// 	benchmark.Flags().StringVarP(&benchmarkStrategy, "strategy", "s", "", "(tps|tps-no-ack) - tps will measure every transaction, tps-no-ack will only measure the last round of transactions")
-// 	benchmark.Flags().BoolVar(&enableJaegerTracing, "jaeger-tracing", false, "enable jaeger tracing")
-// 	benchmark.Flags().BoolVar(&enableElasticTracing, "elastic-tracing", false, "enable elastic tracing")
-// 	benchmark.Flags().StringVar(&tracingTagName, "tracing-name", "", "adds tag to tracing for lookup")
-// }
+	"github.com/quorumcontrol/tupelo-go-sdk/gossip/client"
+	"github.com/quorumcontrol/tupelo-go-sdk/gossip/client/pubsubinterfaces/pubsubwrapper"
+	"github.com/quorumcontrol/tupelo-go-sdk/p2p"
+	"github.com/quorumcontrol/tupelo/gossip/benchmark"
+	"github.com/quorumcontrol/tupelo/nodebuilder"
+	"github.com/spf13/cobra"
+	"gopkg.in/src-d/go-log.v1"
+)
+
+var benchmarkConcurrency int
+var benchmarkTimeout int
+var benchmarkDuration int
+var benchmarkStartDelay int
+
+// benchmark represents the shell command
+var benchmarkCmd = &cobra.Command{
+	Use:    "benchmark",
+	Short:  "runs a set of operations against a network at specified concurrency",
+	Hidden: true,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		config := nodebuilderConfig
+		if config == nil {
+			panic(fmt.Errorf("you must specify a config"))
+		}
+		nb := &nodebuilder.NodeBuilder{Config: config}
+		nb.StartTracing()
+
+		p2pHost, peer, err := p2p.NewHostAndBitSwapPeer(ctx)
+		if err != nil {
+			panic(fmt.Errorf("error creating host: %w", err))
+		}
+
+		if benchmarkStartDelay > 0 {
+			log.Infof("[benchmark] delaying benchmark kickoff for %d seconds...", benchmarkStartDelay)
+			time.Sleep(time.Duration(benchmarkStartDelay) * time.Second)
+		}
+
+		_, err = p2pHost.Bootstrap(config.BootstrapNodes)
+		if err != nil {
+			panic(fmt.Errorf("error bootstrapping: %w", err))
+		}
+
+		group, err := nb.NotaryGroup()
+		if err != nil {
+			panic(fmt.Errorf("error getting notary group: %v", err))
+		}
+
+		if err = p2pHost.WaitForBootstrap(1+len(group.Signers)/2, 15*time.Second); err != nil {
+			panic(err)
+		}
+
+		cli := client.New(group, pubsubwrapper.WrapLibp2p(p2pHost.GetPubSub()), peer)
+		cli.Start(ctx)
+
+		b := benchmark.NewBenchmark(cli, benchmarkConcurrency, time.Duration(benchmarkDuration)*time.Second)
+
+		results := b.Run(ctx)
+
+		out, _ := json.MarshalIndent(results, "", "  ")
+		fmt.Println(string(out))
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(benchmarkCmd)
+	benchmarkCmd.Flags().IntVarP(&benchmarkConcurrency, "concurrency", "c", 1, "how many transactions to execute at once")
+	benchmarkCmd.Flags().IntVarP(&benchmarkDuration, "duration", "d", 10, "how many seconds to run benchmark for")
+	benchmarkCmd.Flags().IntVarP(&benchmarkTimeout, "timeout", "t", 0, "seconds per transaction to wait before timing out")
```

Comment:
Doesn't look like this gets used anywhere. Probably fine to leave it out for now until / unless we need it.